### PR TITLE
Migrate core/marker_manager.js to goog.module syntax

### PR DESCRIPTION
**Pull request by gonfunko**
## The basics

- [x] I branched from `goog_module`
- [x] My pull request is against `goog_module`
- [x] My code follows the [style guide](
      https://developers.google.com/blockly/guides/modify/web/style-guide)
- [x] My code is presented in the form suggested in the [module
      conversion guide](https://github.com/google/blockly/issues/5026)
- [x] I have run `npm test`.

### Resolves

Part of #5026

### Proposed Changes

Converts `core/marker_manager.js` to `goog.module` with ES6 `const`/`let`.

### CODE REVIEW

File: core/marker_manager.js [hunk 1]
```diff
@@ -24,7 +25,7 @@ goog.requireType('Blockly.WorkspaceSvg');
  * @constructor
  * @package
```
Comment (cpcallen) [inline core/marker_manager.js:28]: The current style (see #5026) would have `@package` moved to a separate JSDoc just above the `exports.MarkerManager = …` line.

File: tests/deps.js [hunk 2]
```diff
@@ -107,7 +107,7 @@ goog.addDependency('../../core/keyboard_nav/basic_cursor.js', ['Blockly.BasicCur
 goog.addDependency('../../core/keyboard_nav/cursor.js', ['Blockly.Cursor'], ['Blockly.ASTNode', 'Blockly.Marker', 'Blockly.registry', 'Blockly.utils.object'], {'lang': 'es6', 'module': 'goog'});
 goog.addDependency('../../core/keyboard_nav/marker.js', ['Blockly.Marker'], ['Blockly.ASTNode']);
 goog.addDependency('../../core/keyboard_nav/tab_navigate_cursor.js', ['Blockly.TabNavigateCursor'], ['Blockly.ASTNode', 'Blockly.BasicCursor', 'Blockly.utils.object'], {'lang': 'es6', 'module': 'goog'});
-goog.addDependency('../../core/marker_manager.js', ['Blockly.MarkerManager'], ['Blockly.Cursor', 'Blockly.Marker']);
+goog.addDependency('../../core/marker_manager.js', ['Blockly.MarkerManager'], [], {'lang': 'es6', 'module': 'goog'});
```
Comment (cpcallen) [inline tests/deps.js:110]: Strictly speaking this change should have been included in the previous commit.

**Review by cpcallen — APPROVED** [review]
One small JSDoc nit.  Otherwise LGTM.

File: core/marker_manager.js (hunk 1 above)
Comment (gonfunko) [reply core/marker_manager.js]: Done

File: tests/deps.js (hunk 2 above)
Comment (gonfunko) [reply tests/deps.js:110]: Noted, will double check this for future files.

**Pull request merged**